File: JobAppTracker.Website/Ts/models/jobApp.ts
export class JobApp {
	public Id: number;
	public Title: string;
	public Location: string;
	public Type: string;

	constructor(id: number, title: string, location: string, type: string) {
		this.Id = id;
		this.Title = title;
		this.Location = location;
		this.Type = type;
	}
}

File: JobAppTracker.Website/Models/JobApp.cs
using System.ComponentModel.DataAnnotations;

namespace JobAppTracker.Website.Models
{
	public class JobApp
	{
		public int Id { get; set; }
		public string Title { get; set; }
		public string Location { get; set; }
		public string Type { get; set; }

		[Required]
		public Contact Contact { get; set; }

		[Required]
		public ApplicationUser User { get; set; }
	}
}